### PR TITLE
Added allowAnonymousClones toggle

### PR DESCRIPTION
**Pull request by juwi**
For usage in environments where you don't want to leave the decision about a repo being publicly cloneable to the user, I've added a config flag "allowAnonymousClones" and added the necessary handling for it to RepositoryManager.

### CODE REVIEW

**Comment by gitblit** [comment]
I'm not opposed to this setting but it's needs to be carried-through to completion.

For example, the edit repository page needs to hide the first two access policy choices which allow anonymous clones.

**Comment by juwi** [comment]
Hi,

just wanted to let you know, that I have not forgotten about this. The last weeks have been quite stressful though. I hope I'll get around to adding the UI part on the coming weekend.

**Comment by juwi** [comment]
I finally got around to implementing and testing it. If you can review this again and give me your opinion that'd be great.

File: src/main/java/com/gitblit/wicket/pages/EditRepositoryPage.java [hunk 1]
```diff
@@ -80,6 +80,8 @@
 public class EditRepositoryPage extends RootSubPage {
 
 	private final boolean isCreate;
+	
+	private final boolean allowAnonymousClones;
```
Comment (gitblit) [inline src/main/java/com/gitblit/wicket/pages/EditRepositoryPage.java]: This field is unnecessary in the page.

File: src/main/java/com/gitblit/wicket/pages/NewRepositoryPage.java [hunk 2]
```diff
@@ -64,6 +64,7 @@
 public class NewRepositoryPage extends RootSubPage {
 
 	private final RepositoryModel repositoryModel;
+	private final boolean allowAnonymousClones;
```
Comment (gitblit) [inline src/main/java/com/gitblit/wicket/pages/NewRepositoryPage.java]: This field is unnecessary in the page.

File: src/main/java/com/gitblit/wicket/panels/AccessPolicyPanel.java [hunk 3]
```diff
@@ -48,17 +50,20 @@
...
 
 	private final AjaxFormChoiceComponentUpdatingBehavior callback;
+	
+	private final boolean allowAnonymousClones;
 
 	private RadioGroup<AccessPolicy> policiesGroup;
 
-	public AccessPolicyPanel(String wicketId, RepositoryModel repository) {
-		this(wicketId, repository, null);
+	public AccessPolicyPanel(String wicketId, RepositoryModel repository, boolean allowAnonymousClones) {
```
Comment (gitblit) [inline src/main/java/com/gitblit/wicket/panels/AccessPolicyPanel.java]: This parameter is unnecessary.  You have access to app() here.

File: src/main/java/com/gitblit/wicket/panels/AccessPolicyPanel.java [hunk 4]
```diff
@@ -99,8 +104,10 @@ protected void onInitialize() {
 		if (app().settings().getBoolean(Keys.git.allowAnonymousPushes, false)) {
 			policies.add(anonymousPolicy);
 		}
-		policies.add(authenticatedPushPolicy);
-		policies.add(namedPushPolicy);
+		if (allowAnonymousClones) {
```
Comment (gitblit) [inline src/main/java/com/gitblit/wicket/panels/AccessPolicyPanel.java]:
```java
if (app().settings()....
```

File: src/main/distrib/data/defaults.properties [hunk 5]
```diff
@@ -249,6 +249,16 @@ git.allowCreateOnPush = true
...
 git.allowAnonymousPushes = false
 
+# Global setting to control anonymous clones.
+#
+# This setting allows/rejects anonymous clones at the level of the RepositoryManager.
+# This trumps all repository config settings.  While anonymous clones are convenient
+# on your own box when you are a lone developer,  they are not recommended for
+# any multi-user installation where privacy or security are required.  
+#
+# SINCE Feature request
```
Comment (gitblit) [inline src/main/distrib/data/defaults.properties]: SINCE 1.7.0

File: src/main/distrib/data/defaults.properties [hunk 6]
```diff
@@ -249,6 +249,16 @@ git.allowCreateOnPush = true
...
 
+# Global setting to control anonymous clones.
+#
+# This setting allows/rejects anonymous clones at the level of the RepositoryManager.
+# This trumps all repository config settings.  While anonymous clones are convenient
+# on your own box when you are a lone developer,  they are not recommended for
+# any multi-user installation where privacy or security are required.  
+#
+# SINCE Feature request 
+git.allowAnonymousClones = false
```
Comment (gitblit) [inline src/main/distrib/data/defaults.properties]: Default should be true to be consistent with current behavior.

File: src/main/java/com/gitblit/manager/RepositoryManager.java [hunk 7]
```diff
@@ -904,6 +904,9 @@ private RepositoryModel loadRepositoryModel(String repositoryName) {
 			model.allowForks = getConfig(config, "allowForks", true);
 			model.accessRestriction = AccessRestrictionType.fromName(getConfig(config,
 					"accessRestriction", settings.getString(Keys.git.defaultAccessRestriction, "PUSH")));
+			if (!settings.getBoolean(Keys.git.allowAnonymousClones, true) && !model.accessRestriction.exceeds(AccessRestrictionType.PUSH)) {
```
Comment (gitblit) [inline src/main/java/com/gitblit/manager/RepositoryManager.java]:
```java
&& !model.accessRestriction.atLeast(AccessRestrictionType.CLONE))
```

**Comment by gitblit** [comment]
It would be great if you could squash and rebase on develop.

**Comment by juwi** [comment]
Did a squash into develop on my forked repo.

**Comment by gitblit** [comment]
Ok.  How about the review comments?

**Pull request closed without merging**